runtime/vm.go

```diff
@@ -46,19 +46,26 @@ func NewVM(bytecode *compiler.Bytecode, globals []objects.Object, maxAllocs int6
 	}
 
 	v := &VM{
-		constants:   bytecode.Constants,
+		// constants:   bytecode.Constants,
 		sp:          0,
 		globals:     globals,
 		fileSet:     bytecode.FileSet,
 		framesIndex: 1,
 		ip:          -1,
 		maxAllocs:   maxAllocs,
 	}
+	v.constants = make([]objects.Object, len(bytecode.Constants))
+	for i, c := range bytecode.Constants {
+		v.constants[i] = c
+		if callee, ok := c.(*objects.CompiledFunction); ok {
+			callee.Fn = func(args ...objects.Object) (objects.Object, error) {
```

Comment:
one problem with this is that it directly modifies the `.Fn` attribute of CompiledFunctions in `bytecode.Constants`. What if the same bytecode is used/passed into different VMs?